### PR TITLE
[GRAPE-114] Pytest with GitHub actions

### PR DESCRIPTION
**Pull request by josephkott**
Successful pytest run: https://github.com/genxnetwork/grape/actions/runs/2409472003

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+from reference_directory import ReferenceDirectory
+
+
+HOME_DIRECTORY = os.path.expanduser('~')
+
+GRAPE_DOCKERFILE = 'containers/snakemake/Dockerfile'
+GRAPE_IMAGE_TAG = 'genx_relatives:latest'
+
+REFERENCE_DIRECTORY = os.path.join(HOME_DIRECTORY, 'ref')
+CONTAINER_REFERENCE_DIRECTORY = '/media/ref'
```
Comment (Jahysama) [inline test/test.py:15]: Why do we need 2 reference directories? `REFERENCE_DIRECTORY` looks like `/home/user/ref` while `CONTAINER_REFERENCE_DIRECTORY` is `/media/ref`. First path is used only to initialize `ReferenceDirectory` class instance which later checks for path existence and downloads all files to `CONTAINER_REFERENCE_DIRECTORY` anyway.

Comment (josephkott) [reply test/test.py:15]: For the sake of clarity I just want to split two conceptually different paths: reference directory of the worker machine and the reference directory inside the container.

**Review by Jahysama — APPROVED** [review]
Great work!

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Run Python Tests
+
+on: [workflow_dispatch]
```
Comment (sashaaero) [inline .github/workflows/pytest.yml:3]: Why workflow triggers manually?

File: launcher.py [hunk 3]
```diff
@@ -286,7 +287,7 @@ def get_parser_args():
         raise ValueError('If --impute is present, then --phase must also be present')
 
     if args.command != 'reference' and args.use_bundle:
-        raise ValueError('--bundle option only available for reference downloading')
+        raise ValueError('--use-bundle option only available for reference downloading')
```
Comment (sashaaero) [inline launcher.py:290]: Should it be here?

File: test/test.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+CONTAINER_WORKING_DIRECTORY = '/media/data'
+METRICS_FILEPATH = 'results/metrics.tsv'
+
+
+def _get_download_reference_command(reference_directory):
+    REFERENCE_COMMAND = 'launcher.py reference --use-bundle --ref-directory %s ' \
```
Comment (sashaaero) [inline test/test.py:21]: Not a big issue but f-strings are preferable in modern python.

File: test/test.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            reference_directory.path: {'bind': CONTAINER_REFERENCE_DIRECTORY, 'mode': 'rw'}
+        }
+
+        docker_client.containers.run(GRAPE_IMAGE_TAG, remove=True, command=command, volumes=volumes)
+
+    return reference_directory
+
+
+def test_simulation(docker_client, grape_image, reference_directory):
+    working_directory = os.path.join(HOME_DIRECTORY, 'simulation-ibis')
```
Comment (sashaaero) [inline test/test.py:85]: Why working directory and command are not fixtures?

File: .github/workflows/pytest.yml (hunk 2 above)
Comment (josephkott) [reply .github/workflows/pytest.yml]: Unfortunately test requires a lot of time to run (~3.5h). So, I think we can run them manually when it's really needed. I also added a trigger to run tests when creating a PR to master.

File: launcher.py (hunk 3 above)
Comment (josephkott) [reply launcher.py:290]: Unfortunately yes. I fixed a bug that prevents downloading reference data as bundle.

File: test/test.py (hunk 5 above)
Comment (josephkott) [reply test/test.py:102]: Moved to fixture. I also created a teardown for `working_directory` to remove it after tests.

File: .github/workflows/pytest.yml (hunk 2 above)
Comment (sashaaero) [reply .github/workflows/pytest.yml]: Alright, why do we need github actions then?

**Review by sashaaero — APPROVED** [review]

**Pull request merged**